zio/zngio/scanner.go

```diff
@@ -232,7 +234,11 @@ func (w *worker) wantRecord(rec *zed.Value, progress *zbuf.Progress) bool {
 	// negatives because it expects a buffer of ZNG value messages, and
 	// rec.Bytes is just a ZNG value.  (A ZNG value message is a header
 	// indicating a type ID followed by a value of that type.)
-	if w.filter == nil || w.filter(rec) {
+	// We pass nil to filter for the scope since a scanner cannot appear
+	// inside of a nested scope obviously and buffer filters cannot include
+	// scoped references.
```

Comment:
Is this forward looking or should it be removed?

Reply:
No that's a mistake... will remove.